### ISSUE TITLE
Nested array type leads to compile error

### ISSUE THREAD

**Issue by barnabasJ** [ash-project/ash#501]
**Describe the bug**
Using nested arrays in types leads to a compile error.

**To Reproduce**
```elixir
  calculations do
    calculate :name, :type, Module do
      argument(:test, {:array, {:array, :integer}})
    end
  end
```
**Expected behavior**

It compiles

**Runtime**
 - Elixir version: 1.14.3
 - Erlang version 25.2.1
 - OS Ubuntu 22.04
 - Ash version commit: 54f3a4732600423a6b6108c5de6fc14e7f3373da
 - any related extension versions

**Additional context**

**Comment by zachdaniel** [comment]
Fixed in a043e26116e30f9e6626771327778f9de17f6d19

**Comment by zachdaniel** [comment]
Thanks for the report!
